### DISCUSSION TITLE
My profile password updates successfully, but unable to login with the same password.

### DISCUSSION THREAD

**Discussion by dushmanta05** [Jacobtims/filament-breezy#332] [Q&A] [answered]
Hello Community!

I'm using Breezy for updating my logged in user's details but I've faced an issue where the password is getting updated successfully but when I logged in using the updated password it throws and error of `credentials do not match our records`

The code I've implemented in `Providers\Filament\AdminPanelProvider.php`

```php
->profile(EditProfile::class)
      ->authGuard('web')
      ->plugin(
          BreezyCore::make()
              ->myProfile(
                  shouldRegisterUserMenu: true,
                  shouldRegisterNavigation: false,
                  navigationGroup: 'Settings',
                  hasAvatars: false,
                  slug: 'my-profile'
              )
      )
      ->navigationItems([
          NavigationItem::make('My Profile')
              ->url(fn (): string => route('filament.admin.pages.my-profile'))
              ->isActiveWhen(fn () => request()->routeIs('filament.admin.pages.my-profile'))
              ->icon('heroicon-o-face-smile')
])
```

Here's some screenshot of it

This is my `My Profile` page

![1](https://github.com/jeffgreco13/filament-breezy/assets/73304098/87273d71-0567-4f50-9875-b16d1669d668)

When I update the password, it shows password updates successfully

![2](https://github.com/jeffgreco13/filament-breezy/assets/73304098/f406e4cd-7a36-4ca4-bc42-541e91e00de0)

Then when I log in with the updated password it throws this error

![3](https://github.com/jeffgreco13/filament-breezy/assets/73304098/deac9ba9-65f2-470e-8b40-b7a5b9c532c9)

But when I go and change the password from Super Admin page through the `UserResource` it works.

Screenshot of user password update through Super Admin

![4](https://github.com/jeffgreco13/filament-breezy/assets/73304098/68ee73d2-756f-45da-8dc1-5d07b1273f17)

Also I can confirm the password gets updated, I've checked the `DB` from `phpmyadmin`

I'm stuck with this error could anyone please help me with this. If more details are needed regarding this error please do mention it. Thank you for your time, I really appreciate it. Thank you again.

Also I don't know if it's a bug, so should I open a issue for it?

**Comment by sofyanegi** [comment]
i face same issue, anyone solve maybe?

**Reply by RendyAFS** [reply]
i also to, after create user on filament with` php artisan filamnet:make-user` i can't login, otherwise my email and password is correct

@jeffgreco13 @Jacobtims

**Comment by sofyanegi** [comment]
i make custom page like this

```
<?php

namespace App\Livewire;

use Filament\Forms\Components\TextInput;
use Illuminate\Support\Facades\Hash;
use Illuminate\View\View;
use Jeffgreco13\FilamentBreezy\Livewire\PersonalInfo;

class MyCustomProfile  extends PersonalInfo
{
    public function render() : View
    {
        return view('livewire.my-custom-profile');
    }

    public array $only = ['name', 'email', 'username', 'department', 'companies'];

    protected function getProfileFormComponents(): array
    {
        return $this->getCustomComponents();
    }

    protected function getCustomComponents(): array
    {
        return [
            TextInput::make('name')
                ->label('Name')
                ->required(),

            TextInput::make('email')
                ->label('Email')
                ->email()
                ->required(),

            TextInput::make('username')
                ->label('Username')
                ->required(),

            TextInput::make('department.department_name')
                ->label('Department')
                ->readOnly(),

            TextInput::make('password')
                ->label('Password (optional)')
                ->placeholder('optional')
                ->password()
                ->revealable()
                ->nullable(),
        ];
    }

    public function submit() : void
    {
        $state = $this->form->getState();
        $data = collect($state)->only($this->only)->toArray();

        if (! empty($state['password'])) {
            $this->validate([
                'data.password' => [ 'required'],
            ]);

            $data['password'] = Hash::make($state['password']);
        }

        $this->user->update($data);

        $this->form->fill([
            'name' => $data['name'],
            'email' => $data['email'],
            'username' => $data['username'],
            'department' => $data['department'],
            'password' => null,
        ]);

        $this->sendNotification();
    }
}
```


on AdminPanelProvide.php
```
 ->plugins(plugins: [
                BreezyCore::make()
                    ->myProfileComponents([MyCustomProfile::class])

```

**Comment by RendyAFS** [comment]
Even though I received a successful notification, the password was not updated and is still use the old password. @Jacobtims

**Comment by RendyAFS** [comment, marked as answer]
i think i found it when i change D:\laragon\www\smart-part\vendor\jeffgreco13\filament-breezy\src\Livewire\UpdatePassword.php

on line
```
$data = collect($this->form->getState())->only('new_password')->all();
$this->user->forceFill([
    'password' => Hash::make($data['new_password']),
    'remember_token' => Str::random(60),
]));
```

`to`

```
        
$data = collect($this->form->getState())->only('new_password')->all();
$this->user->forceFill([
    'password' => Hash::make($data['new_password']),
    'remember_token' => Str::random(60),
])->save(); // this one 😁
```

it works.. @Jacobtims can you check and fix the plugin

**Comment by Jacobtims** [comment]
Sorry, I already fixed this but forgot to release it. The fix is now released.
